src/main/java/org/opensearch/eval/App.java

```diff
@@ -142,45 +146,57 @@ public static void main(String[] args) throws Exception {
 
                 } else if(ProbabilityProportionalToSizeQuerySampler.NAME.equalsIgnoreCase(samplerType)) {
 
-                    final ProbabilityProportionalToSizeParametersQuery parameters = gson.fromJson(jsonString, ProbabilityProportionalToSizeParametersQuery.class);
+                    final ProbabilityProportionalToSizeSamplerParameters parameters = gson.fromJson(jsonString, ProbabilityProportionalToSizeSamplerParameters.class);
+                    final ProbabilityProportionalToSizeQuerySampler sampler = new ProbabilityProportionalToSizeQuerySampler(searchEngine, parameters);
+
+                    // TODO: Allow for selecting the queries by date.
+                    final Map<String, Long> querySet = sampler.sample();
+                    querySetId = sampler.indexQuerySet(searchEngine, parameters.getName(), parameters.getDescription(), parameters.getSampling(), querySet);
+
+                } else if(RandomQuerySampler.NAME.equalsIgnoreCase(samplerType)) {
+
+                    final RandomQuerySamplerParameters parameters = gson.fromJson(jsonString, RandomQuerySamplerParameters.class);
+                    final RandomQuerySampler sampler = new RandomQuerySampler(searchEngine, parameters);
 
-                    final ProbabilityProportionalToSizeQuerySampler sampler = new ProbabilityProportionalToSizeQuerySampler(parameters);
                     // TODO: Allow for selecting the queries by date.
-                    final Collection<UbiQuery> ubiQueries = searchEngine.getUbiQueries();
-                    final Map<String, Long> querySet = sampler.sample(ubiQueries);
-                    final String querySetId = sampler.indexQuerySet(searchEngine, parameters.getName(), parameters.getDescription(), parameters.getSampling(), querySet);
+                    final Map<String, Long> querySet = sampler.sample();
+                    querySetId = sampler.indexQuerySet(searchEngine, parameters.getName(), parameters.getDescription(), parameters.getSampling(), querySet);
 
-                    if(querySetId != null) {
-                        System.out.println("Query set created: " + querySetId);
-                    } else {
-                        System.err.println("No queries found for query set.");
-                    }
+                } else if(TopNQuerySampler.NAME.equalsIgnoreCase(samplerType)) {
+
+                    final TopNQuerySamplerParameters parameters = gson.fromJson(jsonString, TopNQuerySamplerParameters.class);
+                    final TopNQuerySampler sampler = new TopNQuerySampler(searchEngine, parameters);
+
+                    // TODO: Allow for selecting the queries by date.
+                    final Map<String, Long> querySet = sampler.sample();
+                    querySetId = sampler.indexQuerySet(searchEngine, parameters.getName(), parameters.getDescription(), parameters.getSampling(), querySet);
```

Comment:
Good old `mapper_parsing_exception` strikes again.

I can imagine it's empty queries again.

Reply:
I need to add some empty queries to the UBI data generator.

Reply:
I added an `empty` to both queries to avoid those.